### ISSUE TITLE
2023年4月8日

### ISSUE THREAD

**Issue by ssfc** [ssfc/wangdao#1145]
211	ubuntu怎样判断conda是否启动？chatGPT: 在Ubuntu中，你可以通过检查终端提示符来判断是否已经启动了conda环境。当你激活了某个conda环境后，终端提示符会发生变化，以显示当前激活的环境名称。例如，如果你已经激活了名为“myenv”的conda环境，那么终端提示符可能会显示类似于以下内容：(myenv) user@ubuntu:~$ 其中，括号内的“myenv”表示当前激活的conda环境的名称。如果你没有激活任何conda环境，则终端提示符将显示默认状态，例如：user@ubuntu:~$ 除了检查终端提示符外，你还可以使用以下命令来列出当前可用的conda环境： conda env list 这将显示所有已创建的conda环境的列表，以及它们的路径和状态信息。已激活的环境将用星号标记。(2023年3月21日)
212	ubuntu conda base环境是什么？chatGPT: 在Ubuntu中，conda base环境是默认创建的conda环境，也是安装Anaconda或Miniconda时自动创建的第一个环境。它位于Anaconda（或Miniconda）的根目录下，并包含了许多常用的科学计算库，如NumPy、SciPy和Matplotlib等。默认情况下，当你打开终端时，Ubuntu会进入base环境。可以通过激活其他conda环境来退出base环境。(2023年3月21日)
213	ubuntu anaconda怎样检查是否安装了pytorch? chatGPT: 要检查在Ubuntu中是否安装了PyTorch，可以使用以下命令：conda list | grep torch 这将列出所有已安装的名字中包含“torch”的包。如果PyTorch已经安装，你应该会看到名为“pytorch”、“torchvision”和“torchaudio”的包。Comment: Ah…果然没有安装. (2023年3月21日)
214	install snap; done; install bottle; done; 这条路走不通, 还是得去github下载二进制文件; (2023年3月21日)
215	Test whether we successfully go to world: curl -x http://127.0.0.1:7890 http://www.google.com/ (2023年3月21日)
216	Comment: it is import for home server to zzz, since we need to upgrade the system and install anaconda; move them to zzz; done; (2023年3月21日)
217	(1) Download clash from github; (2) download Kycloud and rename it config.yaml, put it in the same folder as clash; (3) ./clash-linux-amd64-v3 -d ./ (2023年3月21日) 
218	ubuntu的clash启动后是全局代理，全局翻墙吗？chatGPT: 在 Ubuntu 中启动 Clash 后，默认情况下它是全局代理，即所有网络流量都将通过 Clash 代理服务器进行传输，实现全局翻墙效果。但是，您可以通过编辑 Clash 配置文件来实现更灵活的代理设置，例如只针对特定的应用程序或网站进行代理。// 在 Clash 配置文件中，您可以设置代理规则，包括通过代理的 IP 地址、端口以及不需要代理的 IP 地址、端口。例如，如果您只想代理特定的网站，您可以在配置文件中添加类似以下的规则; (2023年3月21日)
219	Install nvcc on ubuntu; WTF? It costs 5G memory? 怎样查看电脑安装的cuda版本？chatGPT: nvcc: NVIDIA (R) Cuda compiler driver Copyright (c) 2005-2019 NVIDIA Corporation Built on Sun_Jul_28_19:07:16_PDT_2019 Cuda compilation tools, release 10.1, V10.1.243 这个输出表明 CUDA 版本号是 10.1。(2023年3月21日)
220	ubuntu安装clash后，怎样在命令行安装pytorch时走代理信道？chatGPT: 打开终端并输入以下命令，设置系统环境变量以使用代理信道： export http_proxy=http://127.0.0.1:7890 export https_proxy=http://127.0.0.1:7890 // Comment: it works; (2023年3月21日)
221	丫的, cuda没有11.5, 只有11.3; 卸载了重新装吧; (2023年3月21日)
222	Nvidia-smi显示有问题, 难道是驱动没装好？install the recommended driver; 安装完成后都有重启环节; 
223	ubuntu里的ppa源下载速度太慢了，应该怎么解决？ https://www.zhihu.com/question/382334154 : ustc镜像就有ppa的反向代理。每个ppa都会在/etc/apt/sources.list.d 目录下创建一个list文件。只需要将其中的http://ppa.launchpad.net替换为http://launchpad.proxy.ustclug.org就行; Comment: 原来是中科大的镜像. (2023年3月21日)
224	How to install the NVIDIA drivers on Ubuntu 20.04 Focal Fossa Linux How to install the NVIDIA drivers on Ubuntu 20.04 Focal Fossa Linux - Linux Tutorials - Learn Linux Configuration driver   : nvidia-driver-525-open - third-party non-free recommended (Comment: fuck off, do not use open!) 
225	Install proxychains on ubuntu; done; Comment: proxychains is much faster now; (2023年3月21日)
226	我们刚才不该先手安装sudo apt install nvidia-cuda-toolkit, 导致和后面安装的pytorch不兼容了；DONE; (2023年3月21日) 
227	Test whether set seed can reproduce the result; 奇了怪了, 为什么不能? Test whether official code can reproduce or not; 卧槽你这不也不行嘛; 即使设定了相同的随机数种子，是不是若干个epoch后训练结果还是会有所不同？chatGPT: 即使在 PyTorch 中设定了相同的随机数种子，由于一些其他因素（例如，数据的加载顺序或者一些随机性质的正则化技术）的影响，训练结果可能会有一定的差异。因此，在实践中，即使设置了随机数种子，不同的训练运行仍然可能会产生略微不同的结果。(2023年3月22日)
228	torch.save(model, "model.pt")和torch.save({'model_state_dict': model.state_dict(),'optimizer_state_dict': optimizer.state_dict(),}, 'model_weights.pt') 有什么区别？chatGPT: torch.save(model, "model.pt") 会将整个模型对象保存到磁盘上，包括模型的结构、权重、参数等所有信息。这种方法保存的文件通常比较大，适用于需要保存完整模型信息的情况，如模型的部署和推理。而 torch.save({'model_state_dict': model.state_dict(),'optimizer_state_dict': optimizer.state_dict(),}, 'model_weights.pt') 则是只保存模型的权重和优化器的状态字典，这种方式通常会比第一种方式节省磁盘空间，适用于需要保存模型权重以便后续恢复模型参数进行训练的情况。使用第二种方法，我们可以在后续训练时重新创建模型和优化器，然后加载它们的状态字典来恢复之前的训练状态，(2023年3月22日)
229	Learn from chatGPT how to save and load data; create file save_load.py; 测试了, 一样无法在load基础上锐减, 估计是内部的问题, 不用管它了; (2023年3月22日)
230	学会save and load checkpoint; 不知道咋回事, 就是在nn.Embedding上不好使, 算了先这样吧; (2023年3月22日)
231	Check whether np.where tabucol_table is correct or not; 算了反正GCP问题也解决了; (2022年9月9日)
232	找班长学生证注册新学期; 或者自己注册; 已盖章; (2023年3月22日)
233	Rename nodes route in class Route.java; done; rename infoPhe info_pheromone; 奇了怪了, 下文居然没用到? 那删了好了; remove infoPhe; done; (2023年3月22日)
234	Time是什么类型? 原来是自定义的int类型; (2023年3月22日)
235	c++ static_cast是什么意思？chatGPT: 在C++中，static_cast是一种用于执行显式类型转换的运算符。static_cast可以将一种数据类型转换为另一种数据类型，但是只有在转换是安全的情况下才能使用。例如，假设我们有一个float类型的变量x，我们希望将它转换为int类型。我们可以使用static_cast运算符执行此转换：float x = 3.14; int y = static_cast<int>(x); (2023年3月22日)
236	卧槽, 实验室备用的显示器也太烂了, 还没有我笔记本的显示器看得清楚; 另外HDMI线也太短了, 退了吧; Oh, 申请通过了; (2023年3月22日)
237	ChatGPT: hypot() 是一个 C++ 标准库函数，用来计算两个数的平方和的平方根，即 hypot(x, y) = sqrt(x*x + y*y) (2023年3月22日)
238	Compute directly_2023-03-21-14-24-0_window_5_dim_4_epoch_2000_lr_0.001 ; done; (2023年3月22日)
239	c++的类文件名一般大写还是小写？chatGPT: C++的类文件名没有固定的大小写规范，可以根据个人喜好来选择。不过，通常情况下，建议采用大写字母开头的命名方式，这样可以方便地与其他文件进行区分。另外，类文件名一般要和类名保持一致，以便于代码的理解和维护。(2023年3月22日)
